components/raftstore/src/store/peer.rs

```diff
@@ -2727,7 +2727,16 @@ where
                     let commit_index = self.get_store().commit_index();
                     if let Some(read) = self.pending_reads.back_mut() {
                         let max_lease = poll_ctx.cfg.raft_store_max_leader_lease();
-                        if read.propose_time + max_lease > now {
+                        let is_read_index_request = req
+                            .get_requests()
+                            .get(0)
+                            .map(|req| req.has_read_index())
+                            .unwrap_or_default();
+
+                        if !is_read_index_request
```

Comment:
Could you add a comment here that a read index request or a read with addition request always needs the response of checking memory lock, so we cannot apply the optimization here.